applications/FreeSurfaceApplication/python_scripts/edgebased_levelset_solver.py

```diff
@@ -154,76 +151,50 @@ def AddVariables(self) -> None:
         variable_utils.AddDof(KratosMultiphysics.VELOCITY_Y, self.model_part)
         variable_utils.AddDof(KratosMultiphysics.VELOCITY_Z, self.model_part)
 
-
     def ImportModelPart(self) -> None:
         self._ImportModelPart(self.model_part, self.settings["model_import_settings"].Clone())
 
-
     def PrepareModelPart(self) -> None:
         self.model_part.SetBufferSize(self.GetMinimumBufferSize())
 
-
     def Check(self) -> None:
         if all(not component for component in self.body_force):
             raise ValueError("Body force cannot be a zero vector")
 
-
     def Initialize(self) -> None:
         # Get rid of isolated nodes
-        KratosMultiphysics.FindNodalNeighboursProcess(self.model_part).Execute()
+        KratosMultiphysics.FindGlobalNodalNeighboursProcess(self.model_part).Execute()
+        KratosMultiphysics.FindGlobalNodalElementalNeighboursProcess(self.model_part).Execute()
         KratosMultiphysics.EliminateIsolatedNodesProcess(self.model_part).Execute()
 
-        # Initialize remaining nodes
-        initializer = self.density * self.body_force
-        small_value = 1e-4
-        active_node_count = 0
-        for node in self.model_part.Nodes:
-            # Initialize DISTANCE
-            if node.GetSolutionStepValue(KratosMultiphysics.DISTANCE) < 0.0:
-                active_node_count += 1
-                node.SetSolutionStepValue(KratosMultiphysics.DISTANCE, 0, -small_value)
-            else:
-                node.SetSolutionStepValue(KratosMultiphysics.DISTANCE, 0, small_value)
-
-            # Make sure no node has null porosity and diameter
-            # Note: this was set in the main script, not sure if it needs to be here
-            if node.GetSolutionStepValue(KratosMultiphysics.POROSITY) == 0.0:
-                node.SetSolutionStepValue(KratosMultiphysics.POROSITY, 0, 1.0)
-            if node.GetSolutionStepValue(KratosMultiphysics.DIAMETER) == 0.0:
-                node.SetSolutionStepValue(KratosMultiphysics.DIAMETER, 0, 1.0)
-
-            porosity = node.GetSolutionStepValue(KratosMultiphysics.POROSITY)
-            node.SetSolutionStepValue(KratosMultiphysics.PRESS_PROJ, 0, initializer * porosity)
-
-        if not active_node_count:
-            raise RuntimeError("At least 1 node must be initialized with a negative DISTANCE")
-
         # Build edge data structure
         self.matrix_container.ConstructCSRVector(self.model_part)
         self.matrix_container.BuildCSRData(self.model_part)
 
         # Initialize solver
-        self.fluid_solver  = self.__MakeEdgeBasedLevelSet()
-        self.distance_size = 3.0 * self.distance_utils.FindMaximumEdgeSize()
+        self.fluid_solver = self.__MakeEdgeBasedLevelSet()
+
+        if self.use_parallel_distance_calculation:
+            self.distance_size = 3.0 * self.distance_utils.FindMaximumEdgeSize()
+        else:
+            self.distance_size = 3.0 * self.distance_utils.FindMaximumEdgeSize(self.model_part)
```

Comment:
oh good catch; do you know whether there are plans to create a common interface for them?

Reply:
No idea...

Reply:
Indeed, the `FindMaximumEdgeSize` should be removed from the `ParallelDistanceCalculation` as it is only used in here. I'd suggest creating an `EdgeBasedAuxiliaryUtilities` class in the `FreeSurfaceApplication` encapsulating this sort of auxiliary functionalities as static methods.

Reply:
Thanks for the suggestion, @rubenzorrilla. I think I would do it in a future PR.